packages/@lwc/ssr-compiler/src/compile-js/types.ts

```diff
@@ -10,6 +10,16 @@ import type { Node } from 'estree';
 
 export type Visitors = Parameters<typeof traverse<Node, ComponentMetaState>>[1];
 
+export interface WireAdapter {
+    fieldName: string;
+    adapterConstructorId: string;
+    config: Array<{
+        configKey: string;
+        referencedField: string;
+    }>;
+    fieldType: 'property' | 'method';
```

Comment:
TODO: add test for `@wire` on a `get`. We support this:

https://github.com/salesforce/lwc/blob/4845840a8f11a0f0334a7fb50dd82b4f7c2cbe22/packages/%40lwc/integration-types/src/decorators/wire.ts#L235-L239

Reply:
https://github.com/salesforce/lwc/issues/4828